homeassistant/components/octoprint/__init__.py

```diff
@@ -193,8 +193,9 @@ def __init__(self, api_url, key, bed, number_of_tools):
         self.job_last_reading = [{}, None]
         self.job_available = False
         self.printer_available = False
-        self.available = False
         self.printer_error_logged = False
+        self.available = False
+        self.available_error_logged = False
```

Comment:
No need to track this, you can just keep track of `self.available`. If it's set to `False`, log the error, otherwise don't.

Reply:
I've just wanted to log that only once, that's why the `self.available_error_logged` flag was introduced

Reply:
But that's the same as when you only log it when `self.available` _changes_ from True to False? That also happens once, after that all subsequent logging is ignored.

Reply:
I've tried to do it like you've suggested at first, but I found out that there are situations when this won't work, so I've added this extra log guard. When any of those 2 API endpoints(job, printer) is marked as failed before the connection error exception occur, then `self.available` is set to False and the connection error will not be logged. This situation will happen when you have your Octoprint server already running but the printer is offline for example.

If you want I can try look into it more deeply(endpoint and error handling) and make separate PR with changes?

Reply:
This should be handled by a third party library that raises library specific exceptions. We'd welcome PRs that refactor this integration to use a library. 👍

Reply:
Just FYI, I'm half way such a library + integration...